eng/common/docgeneration/Generate-DocIndex.ps1

```diff
@@ -76,16 +76,19 @@ function Get-TocMapping {
             LogWarning "There is no service name for artifact $artifact. Please check csv of Azure/azure-sdk/_data/release/latest repo if this is intended. "
             # If no service name retrieved, print out warning message, and put it into Other page.
             $serviceName = "Other"
-            $displayName = $packageInfo[0].DisplayName.Trim()
         }
         else {
             if ($packageInfo.Length -gt 1) {
                 LogWarning "There are more than 1 packages fetched out for artifact $artifact. Please check csv of Azure/azure-sdk/_data/release/latest repo if this is intended. "
             }
             $serviceName = $packageInfo[0].ServiceName.Trim()
-            $displayName = $packageInfo[0].DisplayName.Trim()
         }
-        $orderServiceMapping[$artifact] = @($serviceName, $displayName)
+        $orderServiceMapping[$artifact] = [PSCustomObject][ordered]@{
+          New = $packageInfo[0].New.ToLower()
+          Type = $packageInfo[0].Type.ToLower()
+          ServiceName = $serviceName
+          DisplayName = $packageInfo[0].DisplayName.Trim()
+       }
```

Comment:
I suggest adding Artifact as a property here as well because that will make sorting a little easier. See my comment below on what I suggest there.

Reply:
$artifact is key of the map. We do sort on this as shown below.
```
  $sortedToc = $tocContent.GetEnumerator() | Sort-Object {
        $_.Value.NewIndex, 
        $_.Value.TypeIndex, 
        $_.Value.ServiceName, 
        $_.Value.DisplayName,
        $_.Key} // This is the artifact
```